### PR TITLE
xclbin version check with tool version

### PR DESCRIPTION
**Pull request by vboggara-xilinx**
xclbin version check with tool version in sw emu and hw emu,

### CODE REVIEW

**Comment by vboggara-xilinx** [comment]
please rebuild

**Comment by abalasa** [comment]
retest this please

File: src/runtime_src/core/pcie/emulation/common_em/config.cxx [hunk 1]
```diff
@@ -821,4 +827,85 @@ namespace xclemulation{
...
+    return strVersionTmp;
+  }
+
+  bool checkXclibinVersionWithTool(const xclBin *header)
+  {
+    bool check = false;
+    auto top = reinterpret_cast<const axlf*>(header);
+    std::string xclbinVersion = xclemulation::getXclbinVersion(top);
+    std::string vivadoVersion = xclemulation::getVivadoVersion();
+    /*if ((vivadoVersion.compare(xclbinVersion)) == 0) {
```
[the remark this answers is not in the document]
Comment (vboggara-xilinx) [reply src/runtime_src/core/pcie/emulation/common_em/config.cxx]: fixed this in code.

File: src/runtime_src/core/pcie/emulation/common_em/config.cxx [hunk 2]
```diff
@@ -821,4 +827,85 @@ namespace xclemulation{
...
+        if (found != std::string::npos) {
+          //std::cout << "Vivado Version : " << strVersion << std::endl;
+          return strVersion;
+        }
+      }
+    }
+    return strVersionTmp;
+  }
+
+  bool checkXclibinVersionWithTool(const xclBin *header)
```
[the remark this answers is not in the document]
Comment (vboggara-xilinx) [reply src/runtime_src/core/pcie/emulation/common_em/config.cxx]: fixed it in code

File: src/runtime_src/core/pcie/emulation/common_em/config.cxx [hunk 3]
```diff
@@ -821,4 +827,85 @@ namespace xclemulation{
...
+    }
+    auto buildMetaData = json_project.get_child("build_metadata");
+    //std::string sTool = buildMetaData.get<std::string>("xclbin.generated_by.name", "");
+    sVersion = buildMetaData.get<std::string>("xclbin.generated_by.version", "");
+    //std::string sTimeStamp = buildMetaData.get<std::string>("xclbin.generated_by.time_stamp", "");    
+    //std::cout << __func__ <<" Tool : " << sTool << " Version : " << sVersion << " TimeStamp : " << sTimeStamp << std::endl;
+    return sVersion;
+  }
+
+  std::string getVivadoVersion()
```
[the remark this answers is not in the document]
Comment (vboggara-xilinx) [reply src/runtime_src/core/pcie/emulation/common_em/config.cxx]: getting vivado install path from the XILINX_VIVADO env variable and searching for string 2XXX.X in the path. if search string matches, it returns that string as vivado version.

**Pull request merged**